notebooks/allen_smFISH.py

```diff
@@ -0,0 +1,201 @@
+
+# EPY: START markdown
+# # Reproduce Allen smFISH results with Starfish
+# 
+# The `allen_smFISH.zip` file needed to follow along with this notebook can be downloaded [here]()
+# 
+# This notebook walks through a work flow that reproduces the smFISH result for one field of view using the starfish package. 
+# It assumes that you have unzipped `allen_smFISH.zip` in the same directory as this notebook. Thus, you should see:
+# 
+# raw/
+# allen_smFISH.ipynb
+# EPY: END markdown
+
+# EPY: START code
+# EPY: ESCAPE %load_ext autoreload
+# EPY: ESCAPE %autoreload 2
+# EPY: END code
+
+# EPY: START code
+from copy import deepcopy
+from glob import glob
+import json
+import os
+
+# EPY: ESCAPE %matplotlib inline
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from scipy import ndimage as ndi
+from scipy import stats
+from skimage import (exposure, feature, filters, io, measure,
+                      morphology, restoration, segmentation, transform,
+                      util, img_as_float)
+
+from starfish.io import Stack
+# EPY: END code
+
+# EPY: START code
+# # developer note: for rapid iteration, it may be better to run this cell, download the data once, and load 
+# # the data from the local disk. If so, uncomment this cell and run this instead of the above. 
+# !aws s3 sync s3://czi.starfish.data.public/allen_smFISH ./allen_smFISH
+# experiment_json = os.path.abspath("./allen_smFISH/fov_001/experiment.json")
+# EPY: END code
+
+# EPY: START code
+# this is a large (1.1GB) FOV, so the download may take some time
+experiment_json = 'https://dmf0bdeheu4zf.cloudfront.net/v0.0.1/allen_smFISH/fov_001/experiment.json'
+# EPY: END code
+
+# EPY: START markdown
+# Load the Stack object, which while not well-named right now, should be thought of as an access point to an "ImageDataSet". In practice, we expect the Stack object or something similar to it to be an access point for _multiple_ fields of view. In practice, the thing we talk about as a "TileSet" is the `Stack.image` object. The data are currently stored in-memory in a `numpy.ndarray`, and that is where most of our operations are done. 
+# 
+# The numpy array can be accessed through Stack.image.numpy\_array (public method, read only) or Stack.image.\_data (read and write)
+# EPY: END markdown
+
+# EPY: START code
+codebook = pd.read_json('https://dmf0bdeheu4zf.cloudfront.net/allen_smFISH/fov_001/codebook.json')
+codebook
+# EPY: END code
+
+# EPY: START markdown
+# We're ready now to load the experiment into starfish (This experiment is big, it takes a few minutes):
+# EPY: END markdown
+
+# EPY: START code
+s = Stack()
+s.read(experiment_json)
+
+# The allen's data is uint, but starfish loads it as floats. Cast it directly to uint16, which does
+# not cause any loss of precision. 
+s.image._data = s.image._data.astype(np.uint16)
```

Comment:
i don't think this is the case any more.

Reply:
You're right. I've removed it. Thanks.